lib/block_view.go

```diff
@@ -2737,27 +2756,28 @@ func (bav *UtxoView) _disconnectAcceptNFTBid(
 
 	// (3) Revert payments made from accepting the NFT bids.
 	if operationData.NFTPaymentUtxoKeys == nil || len(operationData.NFTPaymentUtxoKeys) == 0 {
-		return fmt.Errorf("_disconnectAcceptNFTBid: NFTPaymentUtxoKeys was nil; " +
-			"this should never happen")
+		return fmt.Errorf("NFTPaymentUtxoKeys was nil; this should never happen")
 	}
 	// Note: these UTXOs need to be unadded in reverse order.
 	for ii := len(operationData.NFTPaymentUtxoKeys) - 1; ii >= 0; ii-- {
 		paymentUtxoKey := operationData.NFTPaymentUtxoKeys[ii]
 		if err := bav._unAddUtxo(paymentUtxoKey); err != nil {
-			return errors.Wrapf(err, "_disconnectAcceptNFTBid: Problem unAdding utxo %v: ", paymentUtxoKey)
+			return errors.Wrapf(err, "Problem unAdding utxo %v: ", paymentUtxoKey)
 		}
 	}
 
-	// (4) Revert spent bidder UTXOs.
-	if operationData.NFTSpentUtxoEntries == nil || len(operationData.NFTSpentUtxoEntries) == 0 {
-		return fmt.Errorf("_disconnectAcceptNFTBid: NFTSpentUtxoEntries was nil; " +
-			"this should never happen")
-	}
-	// Note: these UTXOs need to be unspent in reverse order.
-	for ii := len(operationData.NFTSpentUtxoEntries) - 1; ii >= 0; ii-- {
-		spentUtxoEntry := operationData.NFTSpentUtxoEntries[ii]
-		if err := bav._unSpendUtxo(spentUtxoEntry); err != nil {
-			return errors.Wrapf(err, "_disconnectAcceptNFTBid: Problem unSpending utxo %v: ", spentUtxoEntry)
+	if !isNFTBid {
```

Comment:
You can actually just check the UtxoOperation type here.  I would either kill the "isNFTBid" function parameter or add a sanity check to ensure that the UtxoOperation matches the function caller's expectation (ie. make sure that they didn't set isNFTBid to false when the UtxoOperation has OperationTypeNFTBid).  It would also be good to add a nice comment above this if statement to explain why bidder UTXOs don't need to be reverted in the buy-it-now NFT bid case.

Reply:
It also seems good to add an else statement with some sanity checks: (1) sanity check that the operation doesn't have any NFTSpentUtxoEntries if it is an NFT bid, (2) sanity check that the NFT entry has "IsBuyNow" set to true (and you might want to sanity check the opposite case as well).